examples/stepper/stepper.c

```diff
@@ -0,0 +1,151 @@
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+#include <nuttx/motor/stepper.h>
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+struct stepper_args
+{
+  FAR const char * path;
+  int microstep;
+  int speed;
+  int steps;
+};
+
+static int parse_args(int argc, FAR char *argv[],
+                      FAR struct stepper_args *args)
+{
+  if (argc < 3)
+    {
+      return -1;
+    }
+
+  args->path = argv[1];
+  args->microstep = -1;
+  args->speed = 200;
+
+  for (int i = 2; i < argc; ++i)
+    {
+      if (strncmp("-m", argv[i], 2) == 0)
+        {
+          i++;
+          if (i >= argc)
+            {
+              return -1;
+            }
+
+          args->microstep = atoi(argv[i]);
+          i++;
+        }
+
+      if (strncmp("-s", argv[i], 2) == 0)
+        {
+          i++;
+          if (i >= argc)
+            {
+              return -1;
+            }
+
+          args->speed = atoi(argv[i]);
+          i++;
+        }
+    }
+
+  args->steps = atoi(argv[argc - 1]);
+  return 0;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  struct stepper_args args;
+  struct stepper_status_s status;
+  struct stepper_job_s job;
+  int fd;
+  int rc;
+
+  if (parse_args(argc, argv, &args) < 0)
+    {
+      printf("usage:\n"
+            "stepper path\n"
+            "              [-m microstep (sticky)]\n"
+            "              [-s speed in steps/s (default 200)]\n"
+            "              steps (positive: CW, negative: CCW)\n");
+      return 1;
+    }
+
+  fd = open(args.path, O_RDWR);
+  if (fd < 0)
+    {
+      printf("Oops: %s\n", strerror(errno));
+      exit(1);
+    }
+
+  if (args.microstep > 0)
+    {
+      printf("Set microstepping to %d\n", args.microstep);
+      rc = ioctl(fd, STEPIOC_MICROSTEPPING, args.microstep);
+      if (rc < 0)
+        {
+          printf("STEPIOC_MICROSTEPPING: %s\n", strerror(errno));
+          exit(2);
+        }
+    }
+
+  job.steps = args.steps;
+  job.speed = args.speed;
+
+  printf("GO -> %ld @ %ld steps/s\n", job.steps, job.speed);
+
+  /* Move the stepper */
+
+  rc = write(fd, &job, sizeof(struct stepper_job_s)); /* blocking */
+  if (rc != sizeof(struct stepper_job_s))
+    {
+      printf("write: %s\n", strerror(errno));
+    }
+
+  /* Retrieve the absolute position since boot */
+
+  rc = read(fd, &status, sizeof(struct stepper_status_s));
+  if (rc != sizeof(struct stepper_status_s))
+    {
+      printf("read: %s\n", strerror(errno));
+    }
+
+  printf("Current position: %ld\n", status.position);
```

Comment:
Maybe it could be interesting to print the: "Position before stepper motor moving: " and "Position after stepper motor moving: " for diagnostic purposes.